src/commands/projects-create.ts

```diff
@@ -0,0 +1,59 @@
+import * as clc from "cli-color";
+import * as Command from "../command";
+import * as FirebaseError from "../error";
+import { createFirebaseProject, ParentResourceType } from "../firebase-resource-manager";
+import { prompt } from "../prompt";
+import * as requireAuth from "../requireAuth";
+import logger = require("../logger");
+
+module.exports = new Command("projects:create [projectId]")
+  .description("create a new firebase project")
+  .option("-n, --display-name <displayName>", "(optional) display name for the project")
+  .option(
+    "-o, --organization <organizationId>",
+    "(optional) ID of the parent Google Cloud Platform organization under which to create this project"
+  )
+  .option(
+    "-f, --folder <folderId>",
+    "(optional) ID of the parent Google Cloud Platform folder in which to create this project"
+  )
+  .before(requireAuth)
+  .action(async (projectId: string, options: any) => {
+    options.projectId = projectId; // add projectId into options to pass into prompt function
+
+    if (options.organization && options.folder) {
+      throw new FirebaseError(
+        "Invalid argument, please provide only one type of project parent (organization or folder)"
+      );
+    }
+    if (!options.nonInteractive) {
+      await prompt(options, [
+        {
+          type: "input",
+          name: "displayName",
+          default: "",
+          message: "What would you like to call your project? (Press enter to skip)",
+        },
+        {
```

Comment:
`projectId` should be prompted before display name.

Reply:
I have no opposition on this. However there was a conversation in the design proposal between me, @nbegley and Rachel Saunders which agrees that the order of the 2 prompts should match the Firebase Console, which prompts display name first. For me it does make more sense to ask for project id first (since it's mandatory)
@nbegley, How do you think?

Reply:
The experience that users are used to in the Fireconsole is that they input a project display name first, then a unique project ID is suggested to them based off of the name they provided. This lowers friction because the user doesn't have to spend time trying to find a unique project ID, and doesn't have to manually input both "My project" for the name and "my-project-asdf" for the ID.

Does the CLI have a mechanism for pre-populating an input suggestion that the user can then edit? Or is its best equivalent setting a default value and letting the user use that or type a value from scratch?

ie. `Project ID? <my-project-asdf>` vs `Project ID? [my-project-asdf] <>`, where `<...>` is user-editable in the CLI.

Reply:
This is definitely my own personal opinion coming out, but I'm not a fan of the way the console does things here. I care deeply about my project id and couldn't care less about the project name. I always make sure to find a unique project id, then let the name just match it. As far as I'm concerned, any project id with a random hex string at the end is a massive frustration.

It's possible to provide default values for CLI prompts, but doing so based on the previous prompt is a little annoying/tricky.

Reply:
So how do we decide? IMO, I'm on @mbleigh side. Prompt user for `projectId` first seems more intuitive since it's the most important, and we don't need to implement suggestions (which may require some extra api calls to check for uniqueness). Is this a must that we need to follow the practice in Fire Console?

Reply:
I don't feel strongly about this either way, but since this is something Rachel specifically called out in the initial review, let's loop her in (I'll link to here in an email, as she doesn't have a GitHub account).

Reply:
There are pros and cons to each, yeah?

The friction that Nick describes is definitely true about finding a unique Project ID.  The Console currently creates a Project ID with no appended characters if your Project Display Name (currently called "project name") is unique.

But Michael's point is also true that the Console is probably putting too much emphasis on the Project Display Name (which is mutable & internal) rather than on the Project ID (which is immutable AND public-facing).

It's generally best practice in UX if users are presented with consistent decision flows.  In the CLI, things _are_ just a bit different and more complicated cause you're not seeing the Project ID populate as you're typing the Project Display Name.

I'll defer to those that have a strong opinion on this.  Just QQ - Once we set this flow, is it possible to change it later?

Reply:
Nothing set in stone here, we can change it later if we decide to.

Reply:
Thanks, then let's move forward with @mbleigh's suggestion.

Reply:
Updated to prompt `projectId` first